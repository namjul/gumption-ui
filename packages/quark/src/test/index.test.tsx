import * as React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom';
import { quark } from '..';

afterEach(cleanup);

describe('Quark', () => {
  test('renders', () => {
    const Quark = quark('div');
    const { asFragment } = render(<Quark>Hello</Quark>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('render attributes', () => {
    const Quark = quark('div', { attrs: { id: 'foo' } });
    const { asFragment } = render(<Quark className="satori">Hello</Quark>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('render with baseStyle', () => {
    const Quark = quark('div', { baseStyle: { color: 'red' } });
    const { asFragment } = render(<Quark>Hello</Quark>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('render with merged className', () => {
    const Quark = quark('div', { baseStyle: { color: 'red' } });
    const { asFragment } = render(<Quark className="satori">Hello</Quark>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('render `style` attribute', () => {
    const Quark = quark('div');
    const { asFragment } = render(
      <Quark style={{ color: 'red' }}>Hello</Quark>,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test('render with `css` prop', () => {
    const Quark = quark('div');
    const { asFragment } = render(<Quark css={{ color: 'red' }}>Hello</Quark>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('render with `_css` prop', () => {
    const Quark = quark('div');
    const { asFragment } = render(<Quark _css={{ color: 'red' }}>Hello</Quark>);
    expect(asFragment()).toMatchSnapshot();
  });

  test('render with `wrapElement` prop', () => {
    const Quark = quark('div');
    const Anchor = quark('a');
    const { asFragment } = render(
      <Quark wrapElement={(element) => <Anchor>{element}</Anchor>}>
        Hello
      </Quark>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
