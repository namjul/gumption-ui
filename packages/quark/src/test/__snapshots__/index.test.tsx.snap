// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Quark render \`style\` attribute 1`] = `
<DocumentFragment>
  <div
    data-component="div"
    style="color: red;"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`Quark render attributes 1`] = `
<DocumentFragment>
  <div
    class="satori"
    data-component="div"
    id="foo"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`Quark render with \`_css\` prop 1`] = `
<DocumentFragment>
  <div
    class="_hwfcu5"
    data-component="div"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`Quark render with \`css\` prop 1`] = `
<DocumentFragment>
  <div
    class="_hwfcu5"
    data-component="div"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`Quark render with \`wrapElement\` prop 1`] = `
<DocumentFragment>
  <a
    data-component="a"
  >
    <div
      data-component="div"
    >
      Hello
    </div>
  </a>
</DocumentFragment>
`;

exports[`Quark render with baseStyle 1`] = `
<DocumentFragment>
  <div
    class="_hwfcu5"
    data-component="div"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`Quark render with merged className 1`] = `
<DocumentFragment>
  <div
    class="satori _hwfcu5"
    data-component="div"
  >
    Hello
  </div>
</DocumentFragment>
`;

exports[`Quark renders 1`] = `
<DocumentFragment>
  <div
    data-component="div"
  >
    Hello
  </div>
</DocumentFragment>
`;
